dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/relationship/AbstractRelationshipService.java

```diff
@@ -307,8 +307,6 @@ public ImportSummaries updateRelationships( List<Relationship> relationships, Im
             {
                 importSummaries.addImportSummary( updateRelationship( relationship, importOptions ) );
             }
-
-            clearSession();
```

Comment:
why did you remove this? just asking what it was for

Reply:
This was causing a failure in the tests after my changes.
Anyway, we shouldn't clear the session during a transaction. I found out that generally this clearSession() calls were added to make some test pass when more than one call was involved.

The right solution is to clear the session if needed in the test itself between api calls